### PR TITLE
Add preprocessing of Sentinel-1

### LINKED ISSUE #31: Add preprocessing of Sentinel-1

### PR DESCRIPTION
**Pull request by jsreuss**
Closes #31, #38, #42

### CODE REVIEW

File: docs/dataset.md [hunk 1]
```diff
@@ -18,12 +18,16 @@ In order to obtain the observation data for a given country and year, the follow
 
 2.  Clipping of satellite data and calculation of median pixel values.
 
-    {any}`Polygon clipping<eurocropsml.acquisition.clipper>`: Clip parcels from the `.SAFE` files to obtain time series of corresponding reflectance data. As the dataset is intended to be used for crop type classification, we aggregated the collected pixel values. For every parcel and each available time step observation, we calculated the median pixel value for each of the 13 spectral bands of the Sentinel-2 raster tiles, as also done in the [tiny EuroCrops dataset](https://arxiv.org/abs/2106.08151). 
+    {any}`Polygon clipping<eurocropsml.acquisition.clipper>`: Clip parcels from the `.SAFE` files to obtain time series of corresponding reflectance data. As the dataset is intended to be used for crop type classification, we aggregated the collected pixel values. For every parcel and each available time step observation, we calculated the median pixel value for each of the 2 spectral bands of the Sentinel-1 or 13 spectral bands of the Sentinel-2 raster tiles, as also done in the [tiny EuroCrops dataset](https://arxiv.org/abs/2106.08151).
```
Comment (jsreuss) [inline docs/dataset.md:21]:
```suggestion
    {any}`Polygon clipping<eurocropsml.acquisition.clipper>`: Clip parcels from the `.SAFE` files to obtain time series of corresponding reflectance data. As the dataset is intended to be used for crop type classification, we aggregated the collected pixel values. For every parcel and each available time step observation, we calculated the median pixel value for each of the 2 radar bands of the Sentinel-1 or 13 spectral bands of the Sentinel-2 raster tiles, as also done in the [tiny EuroCrops dataset](https://arxiv.org/abs/2106.08151). 
```

File: eurocropsml/dataset/preprocess.py [hunk 2]
```diff
@@ -15,11 +16,15 @@
...
 
 logger = logging.getLogger(__name__)
 
 
+def _unzip_file(zip_filepath: Path, extract_to_path: Path) -> None:
+    with zipfile.ZipFile(zip_filepath, "r") as zip_ref:
+        zip_ref.extractall(extract_to_path)
+
+
 def download_dataset(preprocess_config: EuroCropsDatasetPreprocessConfig) -> None:
```
Comment (jsreuss) [inline eurocropsml/dataset/preprocess.py]: Once we have the final Sentinel-1 data, we should upload them to Zenodo and update this function with the possibility to select whether one wants to download S1 and S2 or only one of the two. I think this should be done before merging any of the current changes.

File: README.md [hunk 3]
```diff
@@ -17,16 +17,16 @@
 
 `EuroCropsML` is a pre-processed and ready-to-use machine learning dataset for crop type classification of agricultural parcels in Europe.
 It consists of a total of **706,683** multi-class labeled data points with a total of **176** distinct classes.
-Each data point contains an annual time series of per parcel median pixel values of Sentinel-2 L1C reflectance data for the year 2021.
+Each data point contains an annual time series of per parcel median pixel values of Sentinel-1 GRD or Sentinel-2 L1C reflectance data for the year 2021.
```
[the remark this answers is not in the document]
Comment (jmaces) [reply README.md:20]: VH should be vertical emission but horizontal reception, no?

[the remark this answers is not in the document]
Comment (jsreuss) [reply README.md:20]: Yes, sorry. That was copy paste :D

File: README.md [hunk 4]
```diff
@@ -17,16 +17,16 @@
...
 For `EuroCropsML`, we acquired and aggregated data for the following countries:
 
-| Country      | Total number of datapoints  | Number of distinct classes | 
-|--------------|--------------------| -------------------------- |
-| Estonia      |  175,906 | 127 |
-| Latvia       | 431,143 | 103 |
-| Portugal     | 99,634 | 79 |
+| Country      | Total number of datapoints for Sentinel-2 | Total number of datapoints for Sentinel-1 | Number of distinct classes | 
+|--------------|-------------------------------------------|-------------------------------------------|----------------------------|
+| Estonia      | 175,906                                   | 154,742                                    | 127                        |
```
Comment (jsreuss) [inline README.md:27]: Note: The number of Sentinel-1 still needs to be updated, once the data has been processed again. This one is missing the MultiPolygons.

File: eurocropsml/dataset/dataset.py [hunk 5]
```diff
@@ -103,12 +102,12 @@ def _format_dates(
...
                     if count == 1:
                         month_data[month] = data[idx]
-                    else:
+                    elif preprocess_config.satellite == "S2":
                         try:
                             cloud_probs = np.apply_along_axis(
                                 partial(
                                     find_clouds,
-                                    band4_idx=EUROCROPS_S2BANDS.index("B4"),
+                                    band4_idx=S2_BANDS.index("04"),
```
Comment (jsreuss) [inline eurocropsml/dataset/dataset.py:110]: Make sure to also adjust the value of `remove_bands` inside the meta-learning repo once this is merged and published.

**Comment by jmaces** [comment]
> **Note:** Once this is merged and published, we need to change the `remove_bands` parameter inside the eurocrops-config (eurocrops.yaml) of the meta-learning repository from `"B10"` to `"10"`.

@jsreuss Since the above note refers to another repository, could you create a ticket for that in the other repo. I feel like it might go unnoticed otherwise, once we close this MR here.

File: README.md [hunk 6]
```diff
@@ -22,11 +22,11 @@ The dataset is based on [Version 9](https://zenodo.org/records/10118572) of [`Eu
...
-| Estonia      | 175,906                                   | 154,742                                    | 127                        |
-| Latvia       | 431,143                                   |                                           | 103                        |
-| Portugal     | 99,634                                    |                                           | 79                         |
+| Country      | Number of distinct classes | Total number of datapoints for Sentinel-2 | Total number of datapoints for Sentinel-1 | 
+|--------------|----------------------------|-------------------------------------------|-------------------------------------------|
+| Estonia      | 127                        | 175,906                                   | 176,055                                   | 
+| Latvia       | 103                        | 431,143                                   | -                                         | 
+| Portugal     | 79                         | 99,634                                    | -                                         |
```
Comment (jmaces) [inline README.md:29]: The total number **706,683** in line 19 above still refers to Sentinel-2 after cloud filtering, correct? Do we want to keep it like that?

File: eurocropsml/acquisition/config.py [hunk 7]
```diff
@@ -11,6 +11,24 @@
...
+    "05",
+    "06",
+    "07",
+    "08",
+    "8A",
+    "09",
+    "10",
+    "11",
+    "12",
+]
```
Comment (jmaces) [inline eurocropsml/acquisition/config.py:28]: Is the order only important for `S1_BANDS` or also for `S2_BANDS`? Maybe we can clarify by having a comment on both lists separately.

Comment (jsreuss) [reply eurocropsml/acquisition/config.py:28]: Good point! It is indeed important for both. I'll add the comment to the S2 bands.

File: README.md (hunk 6 above)
Comment (jsreuss) [reply README.md:29]: Yes, it refers to the final number of Sentinel-2 parcels. I added the number of S1 data points.

**Pull request merged**